### DISCUSSION TITLE
Automatic gui tests on the mac platform

### DISCUSSION THREAD

**Discussion by gunterkoenigsmann** [wxMaxima-developers/wxmaxima#1748] [General]
@wxMaxima-developers/mac-maintainers The maxima-discuss mailing list sometimes sees posts that tell that wxMaxima doesn't work on the new apple silicon macs => I have tried to tell github actions to test to compile wxMaxima to that platform, too - which almost worked: Github actions actually starts a job for a Apple Arm machine - but there seems no apple arm machine, still, that job can run on.

...and I tried to run 
````
ctest
````
after compiling wxMaxima - which executes the first two tests (that don't do anything that involves opening a window) and then halts indefinitely (which seems to indicate that wxMaxima opens a warning or error dialogue that waits for user interaction).
Could anyone owning a mac check if ctest indeed causes a dialogue to open that has to be closed manually? If that is the case and I can get rid of that dialogue perhaps we can get automatic tests of the wxMaxima binary that results from each compilation run.

**Comment by MSoegtropIMC** [comment]
I am using wxMaxima (a meanwhile rather old version from MacPorts) on Apple silicon and Intel silicon and both work fine. So I am not sure what the issue is. How did the reporters install it?

**Reply by gunterkoenigsmann** [reply]
No idea. And I have no idea if the problem is that they just don't trust the old version - or if the new version really doesn't run on the mac, currently.

**Comment by MSoegtropIMC** [comment]
P.S.: running ctest would be a bit of work - I would have to compile outside of MacPorts.

**Reply by gunterkoenigsmann** [reply]
compiling on MacPorts might be something of the likes of

```` 
cd <folder the source is in>
port install gettext ninja pandoc wxWidgets-3.2 maxima gnuplot
mkdir build
cd build
cmake ..
cmake --build
ctest
```` 
But I guess the compilation run will run for hours even if this is correct.

**Reply by MSoegtropIMC** [reply]
> But I guess the compilation run will run for hours even if this is correct.

Well not for hours (MacPorts always compiles it from sources), but a while and that's what I meant with "a bit of effort".

I anyway have it on my todo list to update the MacPorts port - was just busy (well still am, but getting better).

**Comment by MSoegtropIMC** [comment]
Should I try the December 2022 tag or master?

**Reply by gunterkoenigsmann** [reply]
Master - but I found a problem: The wxWidgets version macports can install doesn't come with all the components recent wxMaxima sources require (see https://trac.macports.org/ticket/62904) => If you are willing to test to compile wxMaxima on the mac you'll first have to find out how to make wxWidgets work.
